src/common/indexer/elastic/elastic.indexer.helper.ts

```diff
@@ -761,14 +761,39 @@ export class ElasticIndexerHelper {
   }
 
   buildApplicationFilter(filter: ApplicationFilter): ElasticQuery {
-    let elasticQuery = ElasticQuery.create();
+    let elasticQuery = ElasticQuery.create()
+      .withMustExistCondition('currentOwner');
 
-    if (filter.after) {
-      elasticQuery = elasticQuery.withRangeFilter('timestamp', new RangeGreaterThanOrEqual(filter.after));
+    if (filter.ownerAddress) {
+      elasticQuery = elasticQuery.withMustCondition(QueryType.Match('currentOwner', filter.ownerAddress, QueryOperator.AND));
     }
 
-    if (filter.before) {
-      elasticQuery = elasticQuery.withRangeFilter('timestamp', new RangeLowerThanOrEqual(filter.before));
+    if (filter.addresses !== undefined && filter.addresses.length > 0) {
+      elasticQuery = elasticQuery.withMustMultiShouldCondition(filter.addresses, address => QueryType.Match('address', address));
+    }
+
+    if (filter.search) {
+      elasticQuery = elasticQuery.withSearchWildcardCondition(filter.search, ['address']);
+    }
+
+    if (filter.isVerified !== undefined) {
+      if (filter.isVerified) {
+        elasticQuery = elasticQuery.withMustExistCondition('api_isVerified');
```

Comment:
shouldn't we also check if it is set to `true` ? or we only have `api_isVerified: true` and nothing if `false` ?